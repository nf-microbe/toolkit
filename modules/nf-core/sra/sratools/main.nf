process SRA_SRATOOLS {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/fastp_sra-tools:ce7baca742fa975e' :
        'community.wave.seqera.io/library/fastp_sra-tools:2234a1a896a58751' }"

    input:
    tuple val(meta), val(accession)

    output:
    tuple val(meta), path("${accession}*.fastq.gz") , emit: fastq
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args    = task.ext.args ?: ''
    def args2   = task.ext.args2 ?: ''
    prefix      = task.ext.prefix ?: "${meta.id}"
    """
    ### Download FastQ files with sratools
    prefetch \\
        ${accession} \\
        --output-file ${accession}/${accession}.sra \\
        ${args}

    fasterq-dump \\
        ${accession} \\
        --threads ${task.cpus} \\
        ${args2} \\


    rm -rf ${accession}/${accession}.sra
    gzip ${accession}*.fastq

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sratools: \$(prefetch --version 2>&1 | grep -Eo '[0-9.]+')
    END_VERSIONS
    """

    stub:
    def args    = task.ext.args ?: ''
    prefix      = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${accession}.fastq.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sratools: \$(prefetch --version 2>&1 | grep -Eo '[0-9.]+')
    END_VERSIONS
    """
}
