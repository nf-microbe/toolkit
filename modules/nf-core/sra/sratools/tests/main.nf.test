nextflow_process {

    name "Test Process SRA_SRATOOLS"
    script "../main.nf"
    process "SRA_SRATOOLS"

    tag "modules"
    tag "modules_nfcore"
    tag "sra"
    tag "sra/sratools"

    test("SRR24085752") {
        when {
            process {
                """
                input[0] = [
                    [ id:'SRR24085752', single_end:false, md5_1: 'e67296355980997dde170c602d9cb4c1', md5_2: '86b4ee5d4860aed0a3a2538a7757cbc8' ], // meta map
                    'SRR24085752'
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'SRR24085752', single_end:false, md5_1: 'e67296355980997dde170c602d9cb4c1', md5_2: '86b4ee5d4860aed0a3a2538a7757cbc8' ], // meta map
                    'SRR24085752'
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
