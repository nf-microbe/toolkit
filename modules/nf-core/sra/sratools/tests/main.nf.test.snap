{
    "SRR24085752": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        [
                            "SRR24085752_1.fastq.gz:md5,a1cabbdd6b366eec77afd3afb7b39c7a",
                            "SRR24085752_2.fastq.gz:md5,7487dfa5e3e2a2d0fb34dab9e93c4ae5"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,aaeedb9a859b12f929fbdbbf3e3d87db"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        [
                            "SRR24085752_1.fastq.gz:md5,a1cabbdd6b366eec77afd3afb7b39c7a",
                            "SRR24085752_2.fastq.gz:md5,7487dfa5e3e2a2d0fb34dab9e93c4ae5"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aaeedb9a859b12f929fbdbbf3e3d87db"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-24T13:55:10.698298069"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        "SRR24085752.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,aaeedb9a859b12f929fbdbbf3e3d87db"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        "SRR24085752.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aaeedb9a859b12f929fbdbbf3e3d87db"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-24T13:58:24.989517216"
    }
}